test/helpers/test.go

```diff
@@ -171,6 +177,55 @@ func (t *TestOptions) GetGitProvider() (gits.GitProvider, error) {
 	return gitProvider, nil
 }
 
+// GetApproverGitProvider returns a git provider that uses credentials for the approver user defined in environment variables
+// We don't use standard user auth here because the user/token isn't defined during boot, so we'll create the credentials
+// for the user on the fly.
+func (t *TestOptions) GetApproverGitProvider() (gits.GitProvider, error) {
+	factory := cmd.NewFactory()
+	_, ns, err := factory.CreateKubeClient()
+	if err != nil {
+		return nil, err
+	}
+
+	authConfigService, err := factory.CreateAuthConfigService("gitAuth.yaml", ns, kube.ValueKindGit, "")
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := authConfigService.LoadConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error loading auth config: %s", err)
+	}
+
+	if config == nil {
+		return nil, fmt.Errorf("auth config is nil but no error was returned by LoadConfig")
+	}
+
+	if reflect.DeepEqual(*config, auth.AuthConfig{}) {
```

Comment:
That also feels wrong. Is this needed? What happens if one just calls `config.CurrentAuthServer()`?

Reply:
Beats me? =)